client/src/pages/admin/CreateProduct.test.js

```diff
@@ -0,0 +1,32 @@
+import '@testing-library/jest-dom';
+import React from "react";
+import { render, screen } from '@testing-library/react';
+import { beforeEach, describe, expect } from '@jest/globals';
+import CreateProduct from "./CreateProduct";
+import { BrowserRouter} from "react-router-dom";
+
+const defaultComponent = () => {
+    render(<BrowserRouter>
+        <CreateProduct/>
+    </BrowserRouter>)};
+
+describe("Given a default component is created", () => {
+    // ARRANGE
+    beforeEach(defaultComponent);
+    describe("When the component is rendered", () => {
+        // ACT
+        test("The different components are rendered", () => {
+            // ASSERT
+            expect(screen.getByTitle("Dashboard - Create Product")).toBeInTheDocument(); // admin menu
+            expect(screen.getByPlaceholderText("Select a category")).toBeInTheDocument(); // select category 
+            expect(screen.getByPlaceholderText("Select Shipping ")).toBeInTheDocument(); // select shipping 
+            expect(screen.getByPlaceholderText("write a quantity")).toBeInTheDocument(); // quantity input
```

Comment:
I understand that you are asserting against what was written in the code, but I am not sure whether you should consider the case inconsistencies as a 'bug'. For example, I would have asserted `Write a Quantity` and realised that the code used `write a quantiity`, which is inconsistency in capitalisation.

Reply:
Hi! That's a great point to raise - my current approach to the files I am testing is:

=> Document both cosmetic and functional errors or bugs
=> Any cosmetic bug, I would write the test to match the written version 
=> Any functional bug, I would write the expected behavior into the test with the understanding it would fail

Now, in my second round of unit testing where I improve upon the implemented the tests, expand them and increase coverage, I shall have a chance to revisit these cosmetic bugs and rewrite them such that they fail. Rationale: I would like to view the codebase in its entirety (including other files I have not yet tested) before determining what is the default/expected representation of a component. For e.g., here, I would choose between title case and sentence case after reviewing the codebase and trying to understand the intended implementation.

Hope that helps clarify!

[I will document this strategy - and related bugs - in my section of the report as well]